### PR TITLE
Audio files added as Resource Audio

### PR DESCRIPTION
**Pull request by Rubetoman**
Audio files now are copied to "Library/Audios" and loaded only once. You can view the import configuration using panel browser to load them as "streamed", as well as delete or rename them.

To test (1):
- Load ResourceAudioTest.sc3ne
- Open the ResourceManager tool.
- Load one song on AudioSource GO using the component.
- Check that the audio resource number of references for the file has increased.
- Load the same song on the other AudioSource GO and check that now the number is 2.

To test (2):
- Using the PanelBrowser go to Audio.
- Left click on any audio file.
- Select "Import Configuration"
- Check the checkbox and accept.
- Close the engine.
- Open the engine.
- Using the PanelBrowser go to Audio and left click on the same audio file.
- Select "Import Configuration" and see if the checkbox is checked.

To test (3):
- Using the PanelBrowser go to Audio.
- Left click on any audio file.
- Select "Rename"
- Give it a new name
- Open the ResourceManager tool.
- Check that the resource audio has the new name.

To test (4):
- Using the PanelBrowser go to Audio.
- Left click on any audio file.
- Select "Delete"
- Open the ResourceManager tool.
- Check that the resource audio has been deleted.

### CODE REVIEW

File: Source/ComponentAudioSource.cpp [hunk 1]
```diff
@@ -57,8 +75,16 @@ void ComponentAudioSource::Play()
 	if (enabled) 
 	{
 		Stop();
-		if (!streamed) lastHandler = App->audioManager->PlayWAV(wavFX, Sound3D && !OnlyVolume3D);
-		else lastHandler = App->audioManager->PlayWAV(wavstream, Sound3D && !OnlyVolume3D);
+		if (audio != nullptr)
+		{
+			if (!audio->streamed) lastHandler = App->audioManager->PlayWAV(audio->wavFX, Sound3D && !OnlyVolume3D);
+			else lastHandler = App->audioManager->PlayWAV(audio->wavstream, Sound3D && !OnlyVolume3D);
```
Comment (AlbertVVila) [inline Source/ComponentAudioSource.cpp]: coding style 😞

File: Source/ResourceAudio.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	JSON_value* meta = json->CreateValue();
+	struct stat statFile;
+	stat(filepath.c_str(), &statFile);
+	meta->AddUint("GUID", UID);
+	meta->AddUint("timeCreated", statFile.st_ctime);
+	meta->AddUint("streamed", streamed);
+	json->AddValue("Audio", *meta);
+	filepath += METAEXT;
+	App->fsystem->Save(filepath.c_str(), json->ToString().c_str(), json->Size());
+}
```
Comment (AlbertVVila) [inline Source/ResourceAudio.cpp]: RELEASE json

File: Source/ResourceAudio.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		return;
+	}
+	JSON* json = new JSON(data);
+	JSON_value* value = json->GetValue("Audio");
+
+	streamed = value->GetUint("streamed");
+
+	// Make sure the UID from meta is the same
+	unsigned checkUID = value->GetUint("GUID");
+}
```
Comment (AlbertVVila) [inline Source/ResourceAudio.cpp]: RELEASE json

File: Source/ComponentAudioSource.cpp (hunk 1 above)
Comment (Rubetoman) [reply Source/ComponentAudioSource.cpp]: That is not my code...

**Pull request merged**